ui/pages/permissions-connect/connect-page/utils.ts

```diff
@@ -0,0 +1,154 @@
+function safeParseInt(value: string): number | undefined {
+  const parsed = parseInt(value, 10);
+  return isNaN(parsed) ? undefined : parsed;
+}
+
+/**
+ * Takes in an incoming {@link PermissionsRequest} and attempts to return the {@link Caip25CaveatValue} with the Ethereum accounts set.
+ *
+ * @param permissions - The {@link PermissionsRequest} with the target name of the CAIP-25 endowment permission.
+ * @returns The {@link Caip25CaveatValue} with the Ethereum accounts set. If Caip25 Caveat Type is not found, returns `undefined`.
+ */
+function getCaip25CaveatValue(
+  permissions?: PermissionsRequest,
+): Caip25CaveatValue | undefined {
+  const caip25Permissions = permissions?.[EndowmentTypes.caip25];
+  return caip25Permissions?.caveats?.find(
+    (caveat) => caveat.type === CaveatTypes.caip25,
+  )?.value;
+}
+
+/**
+ * Takes in an incoming {@link PermissionsRequest} and attempts to return the list of requested Ethereum accounts.
+ *
+ * @param permissions - The {@link PermissionsRequest} with the target name of the CAIP-25 endowment permission.
+ * @returns The list of requested Ethereum accounts.
+ */
+export function getRequestedAccountsViaPermissionsRequest(
+  permissions?: PermissionsRequest,
+): string[] {
+  const caip25CaveatValue = getCaip25CaveatValue(permissions);
+  if (!caip25CaveatValue) {
+    return [];
+  }
+  const { optionalScopes } = caip25CaveatValue;
+
+  const allAccountsSet = new Set<string>();
+
+  for (const { accounts } of Object.values(optionalScopes)) {
+    accounts.forEach((accountId: CaipAccountId) =>
+      allAccountsSet.add(parseCaipAccountId(accountId).address),
+    );
+  }
+
+  return Array.from(allAccountsSet);
+}
+
+/**
+ * Takes in an incoming {@link PermissionsRequest} and attempts to return the list of requested chains.
+ *
+ * @param permissions - The {@link PermissionsRequest} with the target name of the CAIP-25 endowment permission.
+ * @returns The list of requested chains.
+ */
+export function getRequestedChainsViaPermissionsRequest(
+  permissions?: PermissionsRequest,
+): string[] {
+  const caip25CaveatValue = getCaip25CaveatValue(permissions);
+  if (!caip25CaveatValue) {
+    return [];
+  }
+
+  const { optionalScopes } = caip25CaveatValue;
+  const result: number[] = [];
+
+  for (const scope of Object.keys(optionalScopes)) {
+    if (scope === 'wallet:eip155') {
+      continue;
+    }
+
+    const { reference } = parseCaipChainId(scope as CaipChainId);
+    const parsedReference = safeParseInt(reference);
+
+    if (parsedReference) {
+      result.push(parsedReference);
+    }
+  }
+
+  return result.map((chainId) => decimalToPrefixedHex(chainId));
+}
+
+/**
+ * Parses the CAIP-25 authorized permissions object after UI confirmation.
+ *
+ * @param addresses - The list of permitted addresses.
+ * @param hexChainIds - The list of permitted chains.
+ * @returns
+ */
+// TODO: advice, should we move this behaviour to @metamask/multichain lib?
```

Comment:
great question, i was wondering the same thing as well. I think this can continue to live in Extension for now. It may become more clear where this should live when we refactor the UI to support non-evm accounts and networks. Open to other thoughts though